permission_handler_web/lib/web_delegate.dart

```diff
@@ -0,0 +1,185 @@
+    switch (webPermissionStatus) {
+      case _grantedPermissionStatus:
+        return PermissionStatus.granted;
+      case _deniedPermissionStatus:
+        return PermissionStatus.permanentlyDenied;
+      case _promptPermissionStatus:
+      default:
+        return PermissionStatus.denied;
+    }
+  }
+
+  Future<PermissionStatus> _permissionStatusState(
+      String webPermissionName, html.Permissions? permissions) async {
+    final webPermissionStatus =
+        await permissions?.query({'name': webPermissionName});
+    return _toPermissionStatus(webPermissionStatus?.state);
+  }
+
+  Future<bool> _requestMicrophonePermission(html.MediaDevices devices) async {
+    html.MediaStream? mediaStream;
+
+    try {
+      mediaStream = await devices.getUserMedia({'audio': true});
+
+      // In browsers, calling [getUserMedia] will start the recording
+      // automatically right after. This is undesired behavior as
+      // [requestPermission] is expected to request permission only.
+      //
+      // The manual stop action is then needed here for to stop the automatic
+      // recording.
+
+      if (mediaStream.active!) {
+        final audioTracks = mediaStream.getAudioTracks();
+        if (audioTracks.isNotEmpty) {
+          audioTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestCameraPermission(html.MediaDevices devices) async {
+    html.MediaStream? mediaStream;
+
+    try {
+      mediaStream = await devices.getUserMedia({'video': true});
+
+      // In browsers, calling [getUserMedia] will start the recording
+      // automatically right after. This is undesired behavior as
+      // [requestPermission] is expected to request permission only.
+      //
+      // The manual stop action is then needed here for to stop the automatic
+      // recording.
+
+      if (mediaStream.active!) {
+        final videoTracks = mediaStream.getVideoTracks();
+        if (videoTracks.isNotEmpty) {
+          videoTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestNotificationPermission() async {
+    bool granted = false;
+    html.Notification.requestPermission().then((permission) => {
+          if (permission == "granted") {granted = true}
+        });
+
+    return granted;
+  }
+
+  Future<PermissionStatus> _requestSingularPermission(
+      Permission permission) async {
+    bool permissionGranted = false;
+
+    switch (permission) {
+      case Permission.microphone:
+        permissionGranted = await _requestMicrophonePermission(devices!);
+        break;
+      case Permission.camera:
+        permissionGranted = await _requestCameraPermission(devices!);
+        break;
+      case Permission.notification:
+        permissionGranted = await _requestNotificationPermission();
+        break;
+      default:
```

Comment:
It would be nice if we could also add support for location permissions. For our Geolocator plugin we have resolved this using the following function:

```dart
Future<bool> _requestLocationPermission() async {
    try {
      final geoPosition = await _geolocation.getCurrentPosition();
      return true;
    } on html.PositionError catch (e) {
      return false;
    }
}
```

The variable `_geolocation` is of type `html.window.navigator.geolocation` and should probably also be supplied as a constructor parameter into this class so the class stays testable.

Reply:
done.